src/coefficients.jl

```diff
@@ -0,0 +1,726 @@
+    number_of_constraints::Int = 0
+
+    constraint_info::Vector{Tuple{DataType,DataType,Int}} =
+        Tuple{DataType,DataType,Int}[]
+
+    matrix_nnz::Int = 0
+
+    matrix_range::Vector{Float64} = sizehint!(Float64[1.0, 1.0], 2)
+    bounds_range::Vector{Float64} = sizehint!(Float64[1.0, 1.0], 2)
+    rhs_range::Vector{Float64} = sizehint!(Float64[1.0, 1.0], 2)
+    objective_range::Vector{Float64} = sizehint!(Float64[1.0, 1.0], 2)
+
+    variables_in_constraints::Set{VariableRef} = Set{VariableRef}()
+    variables_not_in_constraints::Vector{VariableRef} = VariableRef[]
+
+    empty_rows::Vector{ConstraintRef} = ConstraintRef[]
+    bound_rows::Vector{ConstraintRef} = ConstraintRef[]
+    dense_rows::Vector{Tuple{ConstraintRef,Int}} = Tuple{ConstraintRef,Int}[]
+
+    matrix_small::Vector{Tuple{ConstraintRef,VariableRef,Float64}} =
+        Tuple{ConstraintRef,VariableRef,Float64}[]
+    matrix_large::Vector{Tuple{ConstraintRef,VariableRef,Float64}} =
+        Tuple{ConstraintRef,VariableRef,Float64}[]
+
+    bounds_small::Vector{Tuple{VariableRef,Float64}} =
+        Tuple{VariableRef,Float64}[]
+    bounds_large::Vector{Tuple{VariableRef,Float64}} =
+        Tuple{VariableRef,Float64}[]
+
+    rhs_small::Vector{Tuple{ConstraintRef,Float64}} =
+        Tuple{ConstraintRef,Float64}[]
+    rhs_large::Vector{Tuple{ConstraintRef,Float64}} =
+        Tuple{ConstraintRef,Float64}[]
+
+    objective_small::Vector{Tuple{VariableRef,Float64}} =
+        Tuple{VariableRef,Float64}[]
+    objective_large::Vector{Tuple{VariableRef,Float64}} =
+        Tuple{VariableRef,Float64}[]
+
+    has_quadratic_objective::Bool = false
+    has_quadratic_constraints::Bool = false
+    sense::JuMP.OptimizationSense = JuMP.FEASIBILITY_SENSE
+
+    objective_quadratic_range = sizehint!(Float64[1.0, 1.0], 2)
+    matrix_quadratic_range = sizehint!(Float64[1.0, 1.0], 2)
+
+    nonconvex_objective::Bool = false
+    nonconvex_rows::Vector{ConstraintRef} = ConstraintRef[]
+
+    matrix_quadratic_small::Vector{
+        Tuple{ConstraintRef,VariableRef,VariableRef,Float64},
+    } = Tuple{ConstraintRef,VariableRef,VariableRef,Float64}[]
+    matrix_quadratic_large::Vector{
+        Tuple{ConstraintRef,VariableRef,VariableRef,Float64},
+    } = Tuple{ConstraintRef,VariableRef,VariableRef,Float64}[]
+    objective_quadratic_small::Vector{Tuple{VariableRef,VariableRef,Float64}} =
+        Tuple{VariableRef,VariableRef,Float64}[]
+    objective_quadratic_large::Vector{Tuple{VariableRef,VariableRef,Float64}} =
+        Tuple{VariableRef,VariableRef,Float64}[]
+end
+
+function _update_range(range::Vector{Float64}, value::Number)
+    range[1] = min(range[1], abs(value))
+    range[2] = max(range[2], abs(value))
+    return 1
+end
+
+function _get_constraint_data(
+    data,
+    ref::ConstraintRef,
+    func::JuMP.GenericAffExpr;
+    ignore_empty = false,
+)
+    if length(func.terms) == 1
+        if isapprox(first(values(func.terms)), 1.0, rtol = sqrt(eps(Float64)))
```

Comment:
```suggestion
        if first(values(func.terms)) ≈ 1.0
```
or
```suggestion
        if isone(first(values(func.terms)))
```

Reply:
I can go with the first.
But it is currently equivalent to what is written.
The difference is its more robust to changes in Julia.

I dont like isone, becuase someone might be automatically building from reading a file.

This is different than the case for zero which is usually ommited in files, so no issue in reading.

Reply:
> The difference is its more robust to changes in Julia

Julia won't be changing this definition. It would be pretty breaking across the ecosystem